tests/Exception/RequestExceptionTest.php

```diff
@@ -152,4 +152,19 @@ public function testCanProvideHandlerContext()
         $e = new RequestException('foo', $r, null, null, ['bar' => 'baz']);
         $this->assertEquals(['bar' => 'baz'], $e->getHandlerContext());
     }
+
+    public function testObfuscateUrlWithUsername()
+    {
+        $r = new Request('GET', 'http://username@www.oo.com');
+        $e = RequestException::create($r, new Response(500));
+        $this->assertContains('username@', $e->getMessage());
```

Comment:
I would say you should test the whole returned URL here to make sure the rest of the URL is not modified.